### PR TITLE
fix(discover2): Do not overfetch tags

### PR DESCRIPTION
**Pull request by dashed**
## TODO

- [ ] ~delete `Event.getTagsAPIPayload()`~ EDIT: to be deferred in a follow up PR
- [ ] refactor `pickTagParams()` to include `start`, `end`, and `statsPeriod`. Investigate how this was being injected.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/eventsV2/tags.tsx [hunk 1]
```diff
@@ -66,39 +69,54 @@ class Tags extends React.Component<Props, State> {
...
-    return !isAPIPayloadSimilar(thisAPIPayload, otherAPIPayload);
+    return !isEqual(thisAPIPayload, otherAPIPayload);
   };
 
   fetchData = async () => {
     const {api, organization, eventView, location} = this.props;
 
-    this.setState({tags: {}, totalValues: null});
+    const tagsFetchID = Symbol('tagsFetchID');
+    this.setState({tags: {}, totalValues: null, tagsFetchID});
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/tags.tsx]: How does `tagsFetchID` get cleared?

File: src/sentry/static/sentry/app/views/eventsV2/utils.tsx [hunk 2]
```diff
@@ -122,6 +122,20 @@ export type Tag = {
   topValues: Array<TagTopValue>;
 };
 
+type PickTagParamsType = {
+  query: string;
+  field?: string[] | undefined;
+  project?: string | string[] | undefined;
+  sort?: string | string[] | undefined;
+  per_page?: number | undefined;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/utils.tsx]: What about the date range and environment parameters that come from global selection?

File: src/sentry/static/sentry/app/views/eventsV2/tags.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/eventsV2/tags.tsx]: Due to the nature of the requests being performed to populate the tags; it’s not meant to be cleared. The purpose is to have a unique sentinel value associated with the group of API calls made whenever fetchData has a called.

Comment (markstory) [reply src/sentry/static/sentry/app/views/eventsV2/tags.tsx]: :+1: Makes sense. We should re-fetch tag keys when project lists are updated as the available keys will be different.

File: src/sentry/static/sentry/app/views/eventsV2/tags.tsx [hunk 3]
```diff
@@ -35,6 +36,7 @@ type Props = {
 type State = {
   tags: {[key: string]: Tag};
   totalValues: null | number;
+  tagsFetchID: symbol | undefined;
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/eventsV2/tags.tsx]: @dashed change this to be strictly `symbol`.

**Comment by dashed** [comment]
This will be superseded by https://github.com/getsentry/sentry/pull/16326

**Pull request closed without merging**